manifests/vhost.pp

```diff
@@ -2707,7 +2711,64 @@
   # - $ssl_openssl_conf_cmd
   # - $ssl_stapling
   # - $apache_version
-  if $ssl {
+  if $ssl and $ensure == 'present' {
+    if $ssl_reload_on_change {
+      if $ssl_cert {
+        include apache::mod::ssl::reload
+        $_ssl_cert_copy = regsubst($ssl_cert, '/', '_', 'G')
+        file { "${filename}${_ssl_cert_copy}":
+          path    => "${apache::params::puppet_ssl_dir}/${filename}${_ssl_cert_copy}",
+          source  => "file://${ssl_cert}",
```

Comment:
Does this automatically create a dependency on that file or should it have something like:
```puppet
File <| title == $ssl_cert |> -> File["${filename}${_ssl_cert_copy}"]
```

Reply:
We should avoid that and thats the whole point of the way i implemented it: We don't know how the cert file is managed. It may be a file resource in puppet, but does not have to be it. It could be managed completely outside of puppet or via another resource type.
If we would be sure, that the cert file is always a file resource, we could just do a notify dependency from the File to the apache::service and that would be all, but we shouldn't do that.
And yes, that "file://" locator creates a dependency: Once the content of the source file changes, also the "cert copy" file change, resulting into a notify into the apache::service class. This way it works independently of how the real cert files are managed.